lib/inbox/bloc/inbox_bloc.dart

```diff
@@ -255,6 +259,27 @@ class InboxBloc extends Bloc<InboxEvent, InboxState> {
     }
   }
 
+  Future<void> _markAllAsRead(MarkAllAsReadEvent event, emit) async {
+    try {
+      emit(state.copyWith(
+        status: InboxStatus.refreshing,
+      ));
+      Account? account = await fetchActiveProfileAccount();
+      LemmyApiV3 lemmy = LemmyClient.instance.lemmyApiV3;
+
+      if (account?.jwt == null) {
+        return emit(state.copyWith(status: InboxStatus.success));
+      }
+      await lemmy.run(MarkAllAsRead(
+        auth: account!.jwt!,
+      ));
+
+      add(GetInboxEvent(reset: true, showAll: !state.showUnreadOnly));
+    } catch (e) {
+      emit(state.copyWith(status: InboxStatus.failure, errorMessage: e.toString(), totalUnreadCount: 0));
```

Comment:
Just curious about setting the unread count to 0 in case of a failure. Maybe we should just leave it as is? Not sure.

Reply:
Fixed.